### PR TITLE
:seedlings: TK-5002: Add LC Flag Unit Test Cases

### PR DESCRIPTION
**Pull request by pallav-trilio**

### CODE REVIEW

File: cmd/log-collector/cmd/cmd_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			data, err := ioutil.ReadFile(inputFileName)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(yaml.UnmarshalStrict(data, &newLogCollector)).ShouldNot(HaveOccurred())
+
+			Expect(len(newLogCollector.GroupVersionKinds)).ShouldNot(Equal(len(logCollector.GroupVersionKinds)))
+			Expect(len(newLogCollector.LabelSelectors)).ShouldNot(Equal(len(logCollector.LabelSelectors)))
+
+			// check if version populates in pre run if not given
+			Expect(newLogCollector.GroupVersionKinds[0].Version).Should(BeEmpty())
+			Expect(logCollector.GroupVersionKinds[0].Version).ShouldNot(BeEmpty())
```
Comment (msurudwad) [inline cmd/log-collector/cmd/cmd_test.go:87]: also check for exact expected value.. should be `v1` right?

File: cmd/log-collector/cmd/cmd_test.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+			inputFileName = filepath.Join(testDataDir, testInputFileDuplicate)
+			command.PersistentPreRunE(command, []string{})
+			log.Info(logCollector)
+
+			data, err := ioutil.ReadFile(inputFileName)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(yaml.UnmarshalStrict(data, &newLogCollector)).ShouldNot(HaveOccurred())
+
+			Expect(len(newLogCollector.GroupVersionKinds)).ShouldNot(Equal(len(logCollector.GroupVersionKinds)))
+			Expect(len(newLogCollector.LabelSelectors)).ShouldNot(Equal(len(logCollector.LabelSelectors)))
```
Comment (msurudwad) [inline cmd/log-collector/cmd/cmd_test.go:83]: also validate values in these structs

**Review by msurudwad — CHANGES_REQUESTED** [review]

File: cmd/log-collector/cmd/cmd_test.go (hunk 1 above)
Comment (pallav-trilio) [reply cmd/log-collector/cmd/cmd_test.go:99]: done

File: cmd/log-collector/cmd/cmd_test.go (hunk 2 above)
Comment (pallav-trilio) [reply cmd/log-collector/cmd/cmd_test.go:85]: done

File: cmd/log-collector/cmd/cmd_test.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		It("Should throw error for values.yaml file path if provided with incorrect format", func() {
+
+			var newLogCollector logcollector.LogCollector
+			inputFileName = filepath.Join(testDataDir, invalidTestInputFile)
+			pErr := manageFileInputs()
+
+			data, err := ioutil.ReadFile(inputFileName)
+			Expect(err).ShouldNot(HaveOccurred())
+			newLCerr := yaml.UnmarshalStrict(data, &newLogCollector)
+			Expect(pErr.Error()).Should(Equal(newLCerr.Error()))
```
[the remark this answers is not in the document]
Comment (pallav-trilio) [reply cmd/log-collector/cmd/cmd_test.go]: done

[the remark this answers is not in the document]
Comment (pallav-trilio) [reply cmd/log-collector/cmd/cmd_test.go]: done

File: cmd/log-collector/cmd/cmd_test.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+)
+
+var _ = Describe("log collector cmd helper unit tests", func() {
+
+	Context("Initialization of log collector", func() {
+
+		It("Should initialize log collector objects when valid kubeconfig file path is provided", func() {
+
+			command := logCollectorCommand()
+			command.PersistentPreRunE(command, []string{})
```
[the remark this answers is not in the document]
Comment (pallav-trilio) [reply cmd/log-collector/cmd/cmd_test.go]: done

File: cmd/log-collector/cmd/cmd_test.go [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+			Expect(logCollector.CleanOutput).Should(BeFalse())
+		})
+
+		It("Should read values.yaml file path if provided with correct format", func() {
+
+			var newLogCollector logcollector.LogCollector
+			command := logCollectorCommand()
+			inputFileName = filepath.Join(testDataDir, testInputFile)
+			command.PersistentPreRunE(command, []string{})
+			log.Info(logCollector)
```
[the remark this answers is not in the document]
Comment (pallav-trilio) [reply cmd/log-collector/cmd/cmd_test.go]: done

File: .github/workflows/plugin-packages.yml [hunk 6]
```diff
@@ -343,19 +343,19 @@ jobs:
         run: make test-log-collector-plugin-locally
 
       - name: Set up Cloud SDK
-        if: ${{ github.ref == 'refs/heads/main' }}
+        if: (needs.pre-build.outputs.logCollector == 'true') || startsWith(github.ref, 'refs/tags/')
```
Comment (msurudwad) [inline .github/workflows/plugin-packages.yml:346]: `if` not needed .. same condition is already handled at job level

```
  build_test_log-collector:
    runs-on: ubuntu-latest
    needs: pre-build
    # for tags('ref/tags/'), 'needs.pre-build.outputs.logCollector' is 'false' so to run 'build_test_log-collector' job on tags as well,
    # added condition 'startsWith(github.ref, 'refs/tags/')'
    if: (needs.pre-build.outputs.logCollector == 'true') || startsWith(github.ref, 'refs/tags/')
```

File: .github/workflows/plugin-packages.yml [hunk 7]
```diff
@@ -343,19 +343,19 @@ jobs:
...
+          service_account_key: ${{ secrets.GKE_SA_KEY }}
           export_default_credentials: true
 
       - name: Upload Log-Collector binary to GCS artifact
         if: ${{ github.ref == 'refs/heads/main' }}
         run: gsutil cp dist/log-collector_linux_amd64/log-collector gs://trilio-artifacts/tvk-plugins/log-collector/main/log-collector
 
       - name: Authenticate and Connect to GKE cluster
-        if: (needs.pre-build.outputs.testPreflight == 'true') || startsWith(github.ref, 'refs/tags/')
+        if: (needs.pre-build.outputs.logCollector == 'true') || startsWith(github.ref, 'refs/tags/')
```
Comment (msurudwad) [inline .github/workflows/plugin-packages.yml:358]: `if` not needed

File: .github/workflows/plugin-packages.yml (hunk 6 above)
Comment (pallav-trilio) [reply .github/workflows/plugin-packages.yml]: updated

File: .github/workflows/plugin-packages.yml (hunk 7 above)
Comment (pallav-trilio) [reply .github/workflows/plugin-packages.yml]: updated

**Review by msurudwad — APPROVED** [review]

**Pull request merged**